### ISSUE TITLE
[Inspect]Inspect is crashing when moving cursor inside the DataGridView that setting AllowUserToAddRows as false

### ISSUE THREAD

**Issue by Marie-Shi** [dotnet/winforms#4242]
* .NET Core Version: 
.Net 6.0.100-alpha.1.20563.2 from master branch

* Have you experienced this same bug with .NET Framework?: 
Yes

**More info:**
1.	This is a regression issue, it can also **repro** on .Net Framework 4.8, cannot repro on .Net Framework 4.7.2. 

2.	This issue cannot repro when setting AllowUserToAddRows property of DataGridView as true.

**Minimal repro:**
1.	Extract the ‘DataGridView_Inspect_crash.zip’ and open it in VS.
[DataGridView_Inspect_crash.zip](https://github.com/dotnet/winforms/files/5552297/DataGridView_Inspect_crash.zip)

2.	Build and run this application.
3.	Start Inspect and try to focus on the column header of DataGridView, observe the inspect behavior.

**Actual behavior:**
Inspect is crashing when focus on the column header of DataGridView.
![core](https://user-images.githubusercontent.com/45864017/99368341-fc4dfc80-28f5-11eb-9615-9db023a3e2b1.gif)

**Expected behavior:**
Inspect cannot crash and works well when focus on the column header of DataGridView.

**Comment by merriemcgaw** [comment]
@Marie-Shi is there a crash or problem when you use Accessibility Insights for Windows or use Narrator in the same scenario? Just curious :) you've found a high pri bug.

**Comment by Marie-Shi** [comment]
@merriemcgaw Tested this bug with Narrator and Accessibility Insights for Windows, Narrator cannot crash, but for Accessibility Insights for Windows, this tool is **not responding** when moving cursor inside the DataGridView. Please see the below Gif:
![accessibility](https://user-images.githubusercontent.com/45864017/100980519-4d115680-3580-11eb-8ac5-7fcd084db1f6.gif)
> Just curious :) you've found a high pri bug

It is found by a customer. He created a feedback ticket for Framework project.

**Comment by merriemcgaw** [comment]
Can you give me that ticket info? I don't think this meets a servicing bar, but it might. So I'd like to look at it again. Thanks!

**Comment by Marie-Shi** [comment]
@merriemcgaw This is the feedback ticket: [1246370](https://devdiv.visualstudio.com/DevDiv/_workitems/edit/1246370). About this ticket more information, I have send email to you.

**Comment by merriemcgaw** [comment]
I think if we don't see any more feedback from the customer we will just fix it in .NET 6. Please close the feedback ticket and AzDO bug unless the customer pushes back.

**Comment by Marie-Shi** [comment]
@merriemcgaw I have closed the feedback ticket and AzDO bug.

**Comment by tschuebel** [comment]
Hi Marie,

I have just seen you have closed issue 4242.

    In .NET4.8 there is an workaround available:

Workaround is to set Switch.UseLegacyAccessibilityFeatures.3 = true

    In .NET5/6 there is no known workaraund available so a fix is needed. "LegacyAccessibilityFeatures" are removed in .NET5

Please let me know if I can help or support you to fix this this issue.

Best regards, Thomas

**Comment by RussKie** [comment]
@tschuebel could you please provide more information about your use case for us to better understand the impact. Thank you

**Comment by tschuebel** [comment]
**Impact:**
This issue breaks complete ui automation functionality on a WindowsForms Form.

**Situation:**

WindowsForms application containing a DataGridView as follows:
![app](https://user-images.githubusercontent.com/6087871/105544064-db8d9300-5cfa-11eb-9590-c5a07eb14a0d.png)


```cs
public partial class Form1 : Form
{
  /// <summary>
  ///
  /// </summary>
  public Form1()
  {
    InitializeComponent();
    _dataGridView1.Columns.Add("col1", "col1");
    _dataGridView1.Columns.Add("col2", "col2");
  
    _dataGridView1.AllowUserToAddRows = false;
  }
}
```

**Problem**
If a Form contains a DataGridView as shown above its not possible anymore to walk the UITree via:
```cs
private void OnWalkTreeViaUIAInterop_Click(object sender, EventArgs e)
{
    IntPtr hWnd = GetWindowHandle(); // Process main window handle

    if (hWnd != IntPtr.Zero)
    {
        IUIAutomationElement mainWindowAutomationElement = _automation.ElementFromHandle(hWnd);               
        WalkTree(mainWindowAutomationElement, 0);                              
    }
}

private void WalkTree(IUIAutomationElement parent, int indent)
{
    if (parent != null)
    {
        IUIAutomationTreeWalker treeWalker = _automation.ControlViewWalker;
        IUIAutomationElement automationElement = treeWalker.GetFirstChildElement(parent);

        while (automationElement != null)
        {
            for (int x = 0; x <= indent; x++)
            {
                Debug.Write("   ");
            }
            Debug.Write(_controlTypeLookup.GetControlType(automationElement.CurrentControlType) + "(" + automationElement.CurrentName + ")" + "[" + automationElement.CurrentAutomationId + "]" + "\n");

            WalkTree(automationElement, indent + 1);
            IUIAutomationElement next = treeWalker.GetNextSiblingElement(automationElement);
            automationElement = next;
        }
    }
}

```
Walking never stops and produces the repeating output:

**Output:**

      UIA_CustomControlTypeId(Oberste Zeile)[]
         UIA_HeaderControlTypeId(Obere linke Headerzelle)[4236849274]
         UIA_HeaderControlTypeId(col1)[4263208015]
         UIA_HeaderControlTypeId(col2)[4232001227]
      UIA_PaneControlTypeId()[920322]
      UIA_CustomControlTypeId(Oberste Zeile)[]
         UIA_HeaderControlTypeId(Obere linke Headerzelle)[4236849274]
         UIA_HeaderControlTypeId(col1)[4263208015]
         UIA_HeaderControlTypeId(col2)[4232001227]
      UIA_PaneControlTypeId()[920322]
      UIA_CustomControlTypeId(Oberste Zeile)[]
         UIA_HeaderControlTypeId(Obere linke Headerzelle)[4236849274]
         UIA_HeaderControlTypeId(col1)[4263208015]
         UIA_HeaderControlTypeId(col2)[4232001227]
      UIA_PaneControlTypeId()[920322]
      UIA_CustomControlTypeId(Oberste Zeile)[]
         UIA_HeaderControlTypeId(Obere linke Headerzelle)[4236849274]
         UIA_HeaderControlTypeId(col1)[4263208015]
         UIA_HeaderControlTypeId(col2)[4232001227]


On Forms containing many controls there is also a problem with _FindAll(,)_
```cs
IUIAutomationCondition condition = _automation.CreatePropertyCondition(30011, automationId);
IUIAutomationElementArray elementArray = mainWindowAutomationElement.FindAll(TreeScope.TreeScope_Subtree, condition)
```
Specified element could not be found.


If you need more information please let me know.

**Comment by Zheng-Li01** [comment]
Verified the issue with latest 6.0.100-preview.3.21169.6, the issue has been fixed thatI nspect & Accessibility Insights are working well when moving cursor inside the DataGridView that setting AllowUserToAddRows as false.
